controllers/snapshot/snapshot_adapter.go

```diff
@@ -499,6 +407,115 @@ func (a *Adapter) EnsureSnapshotEnvironmentBindingExist() (controller.OperationR
+func (a *Adapter) createEnvironmentBindingForScenario(integrationTestScenario *v1beta1.IntegrationTestScenario, environment *applicationapiv1alpha1.Environment, components *[]applicationapiv1alpha1.Component) (*applicationapiv1alpha1.SnapshotEnvironmentBinding, error) {
+	scenarioLabelAndKey := map[string]string{gitops.SnapshotTestScenarioLabel: integrationTestScenario.Name}
+	binding, err := a.createSnapshotEnvironmentBindingForSnapshot(a.application, environment, a.snapshot, components, scenarioLabelAndKey)
+	if err != nil {
+		a.logger.Error(err, "Failed to create SnapshotEnvironmentBinding for snapshot",
+			"snapshot", a.snapshot.Name,
+			"environment.Name", environment.Name,
+			"snapshot.Spec.Components", a.snapshot.Spec.Components)
+
+		return nil, fmt.Errorf("failed to create SnapshotEnvironmentBinding: %w", err)
+	}
+	a.logger.LogAuditEvent("A snapshotEnvironmentbinding is created", binding, h.LogActionAdd,
+		"integrationTestScenario.Name", integrationTestScenario.Name)
+
+	return binding, nil
+}
+
+// ensureEphemeralEnvironmentForScenarioExists creates or return existing epehemeral environment for the given test scenario
+func (a *Adapter) ensureEphemeralEnvironmentForScenarioExists(integrationTestScenario *v1beta1.IntegrationTestScenario, testStatuses *gitops.SnapshotIntegrationTestStatuses, components *[]applicationapiv1alpha1.Component, allEnvironments *[]applicationapiv1alpha1.Environment) (*applicationapiv1alpha1.SnapshotEnvironmentBinding, error) {
+
+	var (
+		environment *applicationapiv1alpha1.Environment
+		err         error
+	)
+	// environemnt for snapshot
+	environment, ok := a.getExistingEnvironmentForScenario(integrationTestScenario, allEnvironments)
+	if !ok {
+		// environment doesn't exist, we have to create a new one
+		environment, err = a.createEnvironmentForScenario(integrationTestScenario)
+		if err != nil {
+			testStatuses.UpdateTestStatusIfChanged(
+				integrationTestScenario.Name, gitops.IntegrationTestStatusEnvironmentProvisionError,
+				fmt.Sprintf("Creation of ephemeral environment failed: %s", err))
+			a.writeIntegrationTestStatusAtError(testStatuses)
+			return nil, fmt.Errorf("failed to create environment: %w", err)
+		}
+	} else {
+		a.logger.Info("Environment already exists and contains snapshot and scenario:",
+			"environment.Name", environment.Name,
+			"integrationScenario.Name", integrationTestScenario.Name)
+	}
+
+	// envirnoment binding for snapshot
+	binding, err := a.loader.FindExistingSnapshotEnvironmentBinding(a.client, a.context, a.application, environment)
+	if err != nil {
+		a.logger.Error(err, "Failed to find snapshotEnvironmentBinding associated with environment", "environment.Name", environment.Name)
+		return nil, fmt.Errorf("failed to find SnapshotEnvironmentBinding associated with environment: %w", err)
+	}
+	if binding == nil {
+		// create bindging and add scenario name to label of binding
```

Comment:
tiny typo: `bindging` -> `binding`

Reply:
yeah, I just copy pasted it, will change